x-pack/filebeat/module/kubernetes/_meta/fields.yml

```diff
@@ -0,0 +1,134 @@
+- key: kubernetes
+  title: Kubernetes
+  description: >
+    Module for handling logs produced by Kubenetes pods
+  fields:
+  - name: coredns
+    type: group
+    description: >
+      Fields from coredns logs after normalization
+    fields:
+    - name: duration
+      type: double
+
+    - name: client
+      type: ip
+
+    - name: size
+      type: integer
+
+    - name: dnssec_ok
+      type: boolean
+
+    - name: proto
+      type: keyword 
+
+    - name: qclass
+      type: keyword
+
+    - name: rsize
+      type: integer
+
+    - name: log_level
+      type: keyword
+
+    - name: id
+      type: keyword
+
+    - name: rcode
+      type: keyword
+
+    - name: port
+      type: integer
+
+    - name: bufsize
+      type: integer
+
+    - name: qtype
+      type: keyword
+
+    - name: rflags
+      type: keyword
+      description: >
+        coredns response flags
+
+    - name: qname
+      type: keyword
+      description: >
+        coredns query name
+
+    - name: message
+      type: text
+      description: >
+        coredns original log message
+
+  - name: kubernetes
+    type: group
+    description: >
+      Kubernetes meta fields
+    fields:
+    - name: replicaset
+      type: group
+      fields:
+      - name: name
+        type: keyword
+
+  - name: envoy
+    type: group
+    description: >
+      Fields from envoy proxy logs after normalization
+    fields:
+    - name: log_type
+      type: keyword
+      description: >
+        Envoy log type, normally ACCESS
+
+    - name: method
+      type: keyword
+
+    - name: path
+      type: keyword
+
+    - name: proto 
+      type: keyword
+
+    - name: response_code
+      type: integer
+
+    - name: response_flags
+      type: keyword
```

Comment:
Please use ECS fields for http requests and responses https://github.com/elastic/ecs#-http-fields

Reply:
I agree with Jaime's comment here and below. I've added a few more as well.

Is the intent of this module to preserve the original information under `envoy.*`, and only copy out to the ECS fields?

In most places we've simply moved all of the data to the ECS field directly, and not defined the field for the source naming. Of course you still need to define all of the fields that don't actually map to ECS.